### ISSUE TITLE
high CPU and memory usage on failed query

### ISSUE THREAD

**Issue by VikramTiwari** [metabase/metabase#7178]
- Your browser and the version: Chrome latest
- Your operating system: OSX latest
- Your databases: BigQuery
- Metabase version: 0.28.2
- Metabase internal database: MySQL

- *Repeatable steps to reproduce the issue*
A query fails and causes this error to happen:
```
03-22 04:21:28 WARN metabase.query-processor :: {:status :failed,
  :class java.lang.OutOfMemoryError,
  :error "Java heap space",
  :stacktrace [],
  :query
  {:type "native",
   :native
   {:query
    "___redacted_query___"},
   :constraints {:max-results 10000, :max-results-bare-rows 2000},
   :parameters [],
   :cache_ttl nil,
   :info
   {:executed-by 1,
    :context :question,
    :card-id 7001,
    :dashboard-id nil,
    :query-hash [-40, 118, -30, 62, -16, 124, -108, -19, -120, 42, -28, -19, 97, 126, -118, -104, 88, -27, 110, 95, 46, 116, -42, 30, -92, 81, -111, 70, -124, 45, 110, -59],
    :query-type "native"}},
  :expanded-query nil}

 03-22 04:21:36 WARN metabase.query-processor :: Query failure: Java heap space
 ["query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:209)"
  "query_processor$assert_query_status_successful.invoke(query_processor.clj:202)"
  "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:242)"
  "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:235)"
  "query_processor$fn__32020$process_query_and_save_execution_BANG___32025$fn__32026.invoke(query_processor.clj:281)"
  "query_processor$fn__32020$process_query_and_save_execution_BANG___32025.invoke(query_processor.clj:267)"
  "api.card$run_query_for_card.invokeStatic(card.clj:607)"
  "api.card$run_query_for_card.doInvoke(card.clj:593)"
  "api.card$fn__48652$fn__48655$fn__48656.invoke(card.clj:614)"
  "api.card$fn__48652$fn__48655.invoke(card.clj:613)"
  "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:254)"
  "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:249)"
  "api.card$fn__48652.invokeStatic(card.clj:609)"
  "api.card$fn__48652.invoke(card.clj:609)"
  "middleware$enforce_authentication$fn__23041.invoke(middleware.clj:117)"
  "api.routes$fn__52334.invokeStatic(routes.clj:62)"
  "api.routes$fn__52334.invoke(routes.clj:62)"
  "routes$fn__52429$fn__52430.doInvoke(routes.clj:106)"
  "routes$fn__52429.invokeStatic(routes.clj:101)"
  "routes$fn__52429.invoke(routes.clj:101)"
  "middleware$log_api_call$fn__23140$fn__23142.invoke(middleware.clj:353)"
  "middleware$log_api_call$fn__23140.invoke(middleware.clj:352)"
  "middleware$add_security_headers$fn__23090.invoke(middleware.clj:255)"
  "core$wrap_streamed_json_response$fn__52885.invoke(core.clj:67)"
  "middleware$bind_current_user$fn__23045.invoke(middleware.clj:138)"
  "middleware$maybe_set_site_url$fn__23094.invoke(middleware.clj:279)"]
```

**Comment by senior** [comment]
@VikramTiwari Now that we better understand the problem from https://github.com/metabase/metabase/issues/7197, it seems likely that it was the cause of your memory issues. Have you seen this memory issue since you've upgraded?

**Comment by senior** [comment]
I'd be surprised if it fixed it #7191, but it'd be worth testing it out.

Thanks for the update on this one, I'll close it and if it pops up again you can open up a new issue.
